extern/storage-sealing/input.go

```diff
@@ -311,41 +311,45 @@ func (m *Sealing) SectorAddPieceToAny(ctx context.Context, size abi.UnpaddedPiec
 	}
 
 	m.inputLk.Lock()
-	if _, exist := m.pendingPieces[proposalCID(deal)]; exist {
+	if pp, exist := m.pendingPieces[proposalCID(deal)]; exist {
 		m.inputLk.Unlock()
-		return api.SectorOffset{}, xerrors.Errorf("piece for deal %s already pending", proposalCID(deal))
+		select {
+		case <-pp.doneCh:
+			res := pp.resp.Load().(*pieceAcceptResp)
+			return api.SectorOffset{Sector: res.sn, Offset: res.offset.Padded()}, res.err
```

Comment:
Is it safe to call `res.offset.Padded()` if res.err is not nil?

Reply:
Yup it is, it's an uint.